Fabric.Terminology.API/Scripts/Terminology-Install-Utilities.psm1

```diff
@@ -386,6 +386,174 @@ function Publish-TerminologyDacpac() {
+    $dataMartJson = Get-Content -Raw -Path $path
+    $headers = @{"Content-Type" = "application/json"}
+    $headers.Add("Authorization", "Bearer $authToken")
+
+    $mdsUrl = Get-ServiceFromDiscovery -DiscoveryUrl $discoveryServiceUrl -Name "MetadataService" -Version 2
+       
+    try {
+        Write-DosMessage -Level "Information" -Message "Starting to create metadata for $name at " - (get-date).ToString('T')
+        $response = Invoke-RestMethod -Uri $mdsUrl/DataMarts -Method POST -Body $dataMartJson -Headers $headers
+        Write-DosMessage -Level "Information" -Message "Completed creating metadata for $name at " - (get-date).ToString('T')
+        return $response.Id
+    }
+    catch {
+        Write-DosMessage -Level "Error" -Message "Creating metadata stopped for $name at " - (get-date).ToString('T')
+        Write-DosMessage -Level "Error" -Message "Description:" $_.Exception.Response.StatusDescription
+        Write-DosMessage -Level "Error" -Message  $_.ErrorDetails.Message | ConvertFrom-Json | Select-Object -Expand message
+    }
+}
+
+function Post-ToDps {
+    param(
+        [ValidateNotNullOrEmpty()][string] $dataMartId,
+        [ValidateNotNullOrEmpty()][string] $discoveryServiceUrl
+        )
+
+    $headers = @{"Content-Type" = "application/json"}
+    $dpsUrl = Get-ServiceFromDiscovery- DiscoveryUrl $discoveryServiceUrl -Name "DataProcessingService" -version 1
+
+    try {
+        $response = Invoke-RestMethod -Uri $dpsUrl/ExecuteDataMart -Method POST -UseDefaultCredentials -Headers $headers -Body "{ `"DataMartId`": $dataMartId, `"BatchExecution`": { `"PipelineType`": `"Migration`", `"OverrideLoadType`": `"Incremental`", `"LoggingLevel`": `"Diagnostic`" } }"
+        $batchExecutionId = ($response.value | ConvertFrom-Json).Id
+        Write-DosMessage -Level "Info" -Message  "Batch execution successfully sent to the data processing service."
+
+        return $batchExecutionId;
+    }
+    catch {
+        Write-DosMessage -Level "Error" -Message  "POST for '$name' failed with status code:" $_.Exception.Response.StatusCode.value__ 
+        Write-DosMessage -Level "Error" -Message  "Description:" $_.Exception.Response.StatusDescription
+    }
+}
+
+function Poll-BatchExecutions {
+   param(
+        [string] $terminologyBatchExecutionId = $(throw "Please specify the terminologyBatchExecutionId"),
+        [string] $sharedTerminologyBatchExecutionId = $(throw "Please specify the sharedTerminologyBatchExecutionId")
+        )
+
+    $dpsUrl = Get-ServiceFromDiscovery -name "DataProcessingService" -version 1
+    $terminologyUrl = "$dpsUrl/BatchExecutions($terminologyBatchExecutionId)"
+    $terminologyResponse = "";
+    $sharedTerminologyResponse = "";
+    $wasSuccessful = 0;
+
+    foreach($i in 1..30) {
```

Comment:
This runs 30 times no matter what. Shouldn't we exit when $wasSuccessful is true???

Reply:
look at that 😛